framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1371,7 +1375,20 @@ public TransferResult<V, S> visitNarrowingConversion(
     public TransferResult<V, S> visitStringConversion(
             StringConversionNode n, TransferInput<V, S> p) {
         TransferResult<V, S> result = super.visitStringConversion(n, p);
-        result.setResultValue(p.getValueOfSubNode(n.getOperand()));
+
+        Set<AnnotationMirror> operandAnnos = p.getValueOfSubNode(n.getOperand()).getAnnotations();
+        if (operandAnnos.isEmpty() && n.getOperand().getType().getKind() == TypeKind.TYPEVAR) {
+            // use effective annotation if we didn't find any annotations in the value of
+            // a type variable
+            Tree operandTree = n.getOperand().getTree();
+            operandAnnos =
+                    analysis.atypeFactory.getAnnotatedType(operandTree).getEffectiveAnnotations();
```

Comment:
I'm wondering whether this should be `getValueFromFactory` instead of directly calling `getAnnotatedType`. Can you test that?

Reply:
Investigate:

```
import org.checkerframework.checker.nullness.qual.Nullable;
class Demo<S extends Object, T extends S> {
  void foo(T nn, @Nullable T nbl) {
    nbl = nn; // step through this logic
    nbl.toString();
  }
}
```